apps/client/src/features/rundown/event-block/composite/EventBlockChip.tsx

```diff
@@ -0,0 +1,85 @@
+import { useMemo } from 'react';
+import { Tooltip } from '@chakra-ui/react';
+import type { MaybeNumber } from 'ontime-types';
+import { dayInMs, isPlaybackActive, MILLIS_PER_MINUTE, MILLIS_PER_SECOND } from 'ontime-utils';
+
+import { usePlayback, useTimelineStatus } from '../../../../common/hooks/useSocket';
+import { cx } from '../../../../common/utils/styleUtils';
+import { formatDuration } from '../../../../common/utils/time';
+import { tooltipDelayFast } from '../../../../ontimeConfig';
+
+import style from './EventBlockChip.module.scss';
+
+interface EventBlockChipProps {
+  id: string;
+  timeStart: number;
+  isPast: boolean;
+  isLoaded: boolean;
+  className: string;
+  accumulatedGap: MaybeNumber;
+  isNext: boolean;
+  isLinked: boolean;
+  isNextDay: boolean;
+}
+
+export default function EventBlockChip(props: EventBlockChipProps) {
+  const { timeStart, isPast, isLoaded, className, accumulatedGap, isNext, isLinked, isNextDay } = props;
+  const { playback } = usePlayback();
+
+  if (isLoaded) {
+    return null;
+  }
+
+  const playbackActive = isPlaybackActive(playback);
+
+  if (!playbackActive || isPast) {
+    return null; //TODO: Event report will go here
+  }
+
+  if (playbackActive) {
+    return (
+      <EventUntil
+        className={className}
+        timeStart={timeStart + (isNextDay ? dayInMs : 0)}
+        accumulatedGap={accumulatedGap}
+        isNextAndLinked={isNext && isLinked}
+      />
+    );
+  }
+
+  return null;
+}
```

Comment:
question: maybe we dont need the extra element?

at this point, we know that the playback is active (we returned earlier in the guard before) so we could just have the logic of `EventUntil` here

Reply:
hmm can't  use react hooks conditionally I think?

Reply:
ah yes, good point, we should just add a comment for that like
/* we extracted the component to avoid unnecessary calculations and re-renders */